torch/distributed/distributed_c10d.py

```diff
@@ -4555,15 +4605,23 @@ def new_group(
     multiple overlaping process groups. To avoid that, make sure all ranks follow the
     same global creation order.
     """
-    return _new_group_with_tag(
-        ranks,
-        timeout,
-        backend,
-        pg_options,
-        None,
-        use_local_synchronization=use_local_synchronization,
-        group_desc=group_desc,
-    )
+    global _comm_ranks_bucket_cache
+
+    if backend in _communication_cache_enabled_backend:
+        _comm_ranks_bucket_cache[backend] = {}
+        if ranks == None:
+            actual_world_size = get_world_size()
+            ranks_tuple = tuple(list(range(0, actual_world_size)))
```

Comment:
Tuple ctor can also take an iterator in directly

Reply:
done